templates/iam_policy/draft_metadata_validator_lambda_ecr.json.tpl

```diff
@@ -0,0 +1,51 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": [
+        "arn:aws:logs:eu-west-2:${account_id}:log-group:/aws/lambda/tdr-draft-metadata-validator-${environment}",
```

Comment:
A new log group for the new lambda will be created via the `da-terraform-modules`: https://github.com/nationalarchives/da-terraform-modules/blob/main/lambda/main.tf#L83

Therefore the IAM policy will need to be changed to reflect this: `arn:aws:logs:eu-west-2:${account_id}:log-group:/aws/lambda/tdr-draft-metadata-validator-v2-${environment}`

Reply:
I was going to use the same log group - the new lambda will have the same name tdr-draft-metadata-validator-${environment}. terraform destroys the old lambda and creates the new one with the same name but does nothing with the log group

Reply:
At the moment the code in the PR has the functioned name as: `tdr-draft-metadata-validator-v2-${local.environment}` 

Running the Terraform plan a new lambda and CloudWatch log group tare to be to be created with the `v2` name. In which case the permissions in this policy won't work are they are reference the incorrect log group.

Plan snippets:
![Screenshot from 2024-07-19 12-40-59](https://github.com/user-attachments/assets/012f3faa-e12f-45bc-bf24-18b133fc55c0)
![Screenshot from 2024-07-19 12-40-33](https://github.com/user-attachments/assets/046423e7-db8a-441f-b010-ac90d230dab7)